public_html/wp-content/mu-plugins/blocks/assets/src/speakers/block-content.js

```diff
@@ -47,24 +47,28 @@ function SpeakerSessions( { speaker, tracks } ) {
 									{ decodeEntities( session.title.rendered.trim() ) || __( '(Untitled)', 'wordcamporg' ) }
 								</a>
 								<span className="wordcamp-speaker-session-info">
-									{ session.session_track.length &&
+									{ ( session.session_track.length && Array.isArray( tracks ) ) &&
```

Comment:
`[MIGHT]` 🤔 , this feels like kind of a sideways way of handling this. Would it be better to have something more explicit, like this?

```js
function render {
    if ( promise not resolved ) {
        return;
    }
}
```

( or maybe even do it earlier in the lifecycle? )

Reply:
As far as I can tell, when using `withSelect` there isn't a promise exposed anywhere that you can resolve, unfortunately. You just have a prop (tracks in this case) that is `null` until the underlying promise is resolved, at which point it is an array of data.

This might end up being moot if we [move away](https://github.com/WordPress/wordcamp.org/pull/33#discussion_r268608584) from using `withSelect`.

Reply:
We are rethinking the way fetched data is handled in #39, so I'm going to resolve this discussion in favor of that.